Python/bytecodes.c

```diff
@@ -2222,16 +2227,36 @@ dummy_func(
             PyObject *attr_o;
 
             PyDictObject *dict = _PyObject_GetManagedDict(owner_o);
-            DEOPT_IF(hint >= (size_t)dict->ma_keys->dk_nentries);
+            DEOPT_IF(!LOCK_OBJECT(dict));
```

Comment:
I'm not thrilled with the lock here -- we've tried to avoid locking in the fast path dictionary accesses -- but given the overall perf gains, we can come back to this later.